app/scripts/contentscript.js

```diff
@@ -42,7 +42,12 @@ function injectScript(content) {
     const container = document.head || document.documentElement;
     const scriptTag = document.createElement('script');
     scriptTag.setAttribute('async', 'false');
-    scriptTag.textContent = content;
+    // Inline scripts do not work in MV3 due to more strict security policy
+    if (process.env.ENABLE_MV3) {
+      scriptTag.setAttribute('src', browser.runtime.getURL('inpage.js'));
```

Comment:
this will make the ethereum api injection late.
i think theres a new mv3 way of doing this, mentioned in this thread https://bugs.chromium.org/p/chromium/issues/detail?id=634381#c38
```suggestion
      // for sync injection see https://bugs.chromium.org/p/chromium/issues/detail?id=634381#c38
      scriptTag.setAttribute('src', browser.runtime.getURL('inpage.js'));
```
we should at least link to this thread in a comment
```js
if (_browser.runtime.getManifest().manifest_version == 3) {
    _browser.scripting.unregisterContentScripts().then(() => {
        var scripts = [{
             id: "inject",
             js: ["inject.js"],
             matches: ["<all_urls>"],
             world: "MAIN",
             runAt: "document_start",
         }];
        _browser.scripting.registerContentScripts(scripts);
    });
```

Reply:
Documents say that MV3 support this, but I could not get this to work:

![Screenshot 2022-05-25 at 7 52 36 AM](https://user-images.githubusercontent.com/2182307/170166008-18ec1a39-ca3a-43e8-a42d-78a36a383dd5.png)